pkg/kube/inject/app_probe.go

```diff
@@ -187,42 +192,60 @@ func DumpAppProbers(pod *corev1.Pod, targetPort int32) string {
 		}
 
 		if probePort.Type == intstr.String {
-			port, exists := portMap[probePort.StrVal]
-			if !exists {
+			// First, check if the string port refers to a named port in the specific container
```

Comment:
1. Skip this if neither include or exclude annotations are present.
2. Refactor named port conversion to integer to another function.

Reply:
1. the named port logic still needs to execute regardless of whether annotations are present or not.
2. Done.